packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1530,6 +1542,15 @@ function applyVirtualResolutionMutations({
 
           if (peerDescriptor.range === `missing:`) {
             missingPeerDependencies.add(peerDescriptor.identHash);
+          } else if (report !== null) {
+            // When the parent provides the peer dependency request it must be checked to ensure
+            // it is a compatible version.
+            const peerPackage = getPackageFromDescriptor(peerDescriptor);
+            if (!semver.validRange(peerRequest.range)) {
+              report.reportWarning(MessageName.INVALID_RANGE_PEER_DEPENDENCY, `${structUtils.prettyLocator(project.configuration, pkg)} requests peer dependency ${structUtils.prettyDescriptor(project.configuration, peerRequest)} which is not a valid semver range. Cannot ensure peer accuracy with the provided package ${structUtils.prettyLocator(project.configuration, peerPackage)}`);
+            } else if (peerPackage.version === null || !semver.satisfies(peerPackage.version, peerRequest.range, {includePrerelease: peerRequest.range === '*'})) {
```

Comment:
The semver check uses [includePrerelease](https://www.npmjs.com/package/semver#functions) for * ranges because without it peer dependencies that list * do include pre release versions like "0.1.0-beta.10". In the berry repo without this edge case would show over 50+ warnings for typescript because it is currently on a pre release version. This does make the assumption that * indicates any version including pre release. I'm currently checking behavior of yarn v1 and npm (should have done before PR was put up)

Reply:
yarn did not yell about pre release versions when using *

Reply:
The v1 has a [fairly complicated logic](https://github.com/yarnpkg/yarn/blob/master/src/util/semver.js#L10) to support prereleases - which was built in a time before `includePrerelease` even existed 😄